packages/drawer/rc/Drawer.js

```diff
@@ -0,0 +1,156 @@
+import { css } from 'glamor'
+import PropTypes from 'prop-types'
+import React, { forwardRef, useEffect, useImperativeHandle } from 'react'
+
+import filterReactProps from '@pluralsight/ps-design-system-filter-react-props'
+import { CaretDownIcon } from '@pluralsight/ps-design-system-icon'
+import { useTheme } from '@pluralsight/ps-design-system-theme'
+import { useCollapsible } from '@pluralsight/ps-design-system-collapsible/rc'
+
+import stylesheet from './css/index.js'
+
+import { DrawerContext, useDrawerContext } from './context.js'
+import { actionTypes, useDrawer } from './use-drawer.js'
+import { usePrevious } from './use-previous.js'
+import { combineFns, isDefined, isFunction } from './utils.js'
+
+const styles = {
+  head: (themeName, { isOpen }) =>
+    css(
+      stylesheet[`.psds-drawer__head`],
+      isOpen && stylesheet['.psds-drawer__head.psds-drawer--isOpen'],
+      stylesheet[`.psds-drawer__head.psds-theme--${themeName}`]
+    ),
+  body: themeName =>
+    css(
+      stylesheet['.psds-drawer__body'],
+      stylesheet[`.psds-drawer__body.psds-theme--${themeName}`]
+    ),
+  iconSlot: () => css(stylesheet['.psds-drawer__icon-slot']),
+  rotatable: (themeName, { isOpen }) =>
+    css(
+      stylesheet['.psds-drawer__rotatable'],
+      stylesheet[`.psds-drawer__rotatable.psds-theme--${themeName}`],
+      isOpen && stylesheet['.psds-drawer__rotatable.psds-drawer--isOpen']
+    ),
+  collapsible: () => css(stylesheet['.psds-drawer__collapsible'])
+}
+
+const Drawer = forwardRef((props, ref) => {
+  const { children, onChange, stateReducer, ...rest } = props
+
+  const controlled = isDefined(props.isOpen)
+
+  const initialState = {
+    ...props.initialState,
+    ...(controlled && { controlled, isOpen: props.isOpen })
+  }
+
+  const context = useDrawer({
+    initialState,
+    stateReducer,
+
+    onRequestClose: props.onRequestClose,
+    onRequestOpen: props.onRequestOpen
```

Comment:
a couple nice event functions that be used to build a controlled component without going full on state reducer